apps/tenant-management-webapp/src/app/pages/admin/services/form/definitions/addEditFormDefinitionEditor.tsx

```diff
@@ -0,0 +1,368 @@
+import React, { FunctionComponent, useState, useEffect, useRef } from 'react';
+
+import Editor from '@monaco-editor/react';
+import { FormDefinition } from '@store/form/model';
+
+import { toKebabName } from '@lib/kebabName';
+import { useValidators } from '@lib/validation/useValidators';
+import { isNotEmptyCheck, wordMaxLengthCheck, badCharsCheck, duplicateNameCheck } from '@lib/validation/checkInput';
+import { FETCH_KEYCLOAK_SERVICE_ROLES } from '@store/access/actions';
+import { ActionState } from '@store/session/models';
+import { ClientRoleTable } from '@components/RoleTable';
+import {
+  SpinnerModalPadding,
+  FormFormItem,
+  HelpText,
+  DescriptionItem,
+  ErrorMsg,
+  TextLoadingIndicator,
+  FlexRow,
+  FlexLeft,
+  FlexRight,
+  EditorPadding,
+  FinalButtonPadding,
+} from '../styled-components';
+import { GoAPageLoader } from '@abgov/react-components';
+import { FetchRealmRoles } from '@store/tenant/actions';
+import { ConfigServiceRole } from '@store/access/models';
+import { getFormDefinitions } from '@store/form/action';
+import { updateFormDefinition } from '@store/form/action';
+
+import { createSelector } from 'reselect';
+
+import { RootState } from '@store/index';
+import { useSelector, useDispatch } from 'react-redux';
+import { fetchKeycloakServiceRoles } from '@store/access/actions';
+import { defaultFormDefinition } from '@store/form/model';
+
+import { useHistory, useParams } from 'react-router-dom';
+
+import { GoATextArea, GoAInput, GoAButtonGroup, GoAFormItem, GoAButton, GoAIcon } from '@abgov/react-components-new';
+
+function getWindowDimensions() {
+  const { innerWidth: width, innerHeight: height } = window;
+  return {
+    width,
+    height,
+  };
+}
+
+export default function useWindowDimensions() {
+  const [windowDimensions, setWindowDimensions] = useState(getWindowDimensions());
+
+  useEffect(() => {
+    function handleResize() {
+      setWindowDimensions(getWindowDimensions());
+    }
+
+    window.addEventListener('resize', handleResize);
+    return () => window.removeEventListener('resize', handleResize);
+  }, []);
+
+  return windowDimensions;
+}
+
+export const AddEditFormDefinitionEditor: FunctionComponent = () => {
```

Comment:
I would suggest using the JSX.Element here. We shall try to deprecate unnecessary functionComponent.